src/mqt/bench/devices/quantinuum.py

```diff
@@ -11,14 +11,41 @@
 from __future__ import annotations
 
 import json
-from typing import TYPE_CHECKING, TypedDict, cast
+from typing import TypedDict, cast
 
-if TYPE_CHECKING:
-    from pathlib import Path
+from .calibration import DeviceCalibration, get_device_calibration_path
+from .device import Device, Gateset
 
-from .calibration import DeviceCalibration
-from .device import Device
-from .provider import Provider
+
+class QuantinuumH2(Device):
+    """Quantinuum H2 device."""
+
+    def __init__(self) -> None:
+        """Initialize the Quantinuum device."""
+        with get_device_calibration_path("quantinuum_h2").open() as json_file:
+            self.quantinuum_calibration = cast("QuantinuumCalibration", json.load(json_file))
+        self.calibration = None
+
+        self.name = self.quantinuum_calibration["name"]
+        self.gateset = Gateset("quantinuum", self.quantinuum_calibration["basis_gates"])
+        self.num_qubits = self.quantinuum_calibration["num_qubits"]
+        self.coupling_map = list(self.quantinuum_calibration["connectivity"])
```

Comment:
When we switch to Qiskit's Target for representing devices, we can still use JSON files for representing the devices. We just need to follow IBMs format.

This is probably a bit out of place here, but we probably also want to vendor the Qiskit Runtime-provided fake backend data as part of MQT Bench so that we can remove the ibm-Qiskit-runtime dependency from MQT Bench for now.